policy-controller/k8s/index/src/index.rs

```diff
@@ -902,7 +903,16 @@ impl Pod {
                         continue;
                     }
 
-                    let s = policy.inbound_server(srvname.clone(), server, authentications);
+                    let s = policy.inbound_server(
+                        srvname.clone(),
+                        server,
+                        authentications,
+                        self.probes
+                            .get(&port)
+                            .into_iter()
+                            .flatten()
+                            .map(|p| p.as_str()),
```

Comment:
tiny style nit, definitely unimportant: the same code is duplicated here: https://github.com/linkerd/linkerd2/blob/3a34f9734a79c159d98660cf9098df97b13bcbd2/policy-controller/k8s/index/src/index.rs#L1060
and this is kind of wordy for what it's doing...it _might_ be worth factoring it out into something like
```rust
impl Pod {
    // ...

    fn probes_for_port(&self, port: NonZeroU16) -> impl Iterator<Item  = &str> {
        self.probes.get(&port).into_iter().flatten().map(String::as_str)
    }
}
```
which we could then just call here
```suggestion
                        self.probes_for_port(port),
```
and we could change `default_inbound_server` to take `probes` as an `impl Iterator<Item = &str>` as well.

this isn't a big deal at all, and IMO the current code is totally fine, but we could maybe get away without having to write out the whole `.get(...).into_iter().flatten()` thing twice...

Reply:
Yea unfortunately this doesn't work out too well because `fn port_server_or_default` borrows `&mut self`. This means that we can't introduce `fn probes_for_port` and call it in that function because we've already mutably borrowed `self`.

Reply:
oh, i see, that's kind of a bummer. i suppose a `probes_for_port` method could be implemented _on the `PortMap`, so that we don't have to borrow the whole `Pod` in that method, but at that point we are probably overcomplicating things, so, yeah, this is fine.